mmf/models/frcnn.py

```diff
@@ -0,0 +1,555 @@
+        return x
+
+    @property
+    def out_channels(self):
+        return self.conv1.out_channels
+
+    @property
+    def stride(self):
+        return 4  # = stride 2 conv -> stride 2 max pool
+
+
+class GeneralizedRCNN(nn.Module):
+    def __init__(self, cfg):
+        super().__init__()
+
+        self.device = torch.device(cfg.MODEL.DEVICE)
+        self.backbone = build_backbone(cfg)
+        self.proposal_generator = RPN(cfg, self.backbone.output_shape())
+        self._fix_proposal_generator(cfg)
+        self.roi_heads = Res5ROIHeads(cfg, self.backbone.output_shape())
+        self._fix_res5_block(cfg)
+        self.to(self.device)
+
+    def _fix_proposal_generator(self, cfg):
+        in_channels = [
+            val.channels for key, val in self.backbone.output_shape().items()
+        ]
+        assert len(set(in_channels)) == 1, "Each level must have the same channel!"
+        in_channels = in_channels[0]
+        if cfg.MODEL.PROPOSAL_GENERATOR.HIDDEN_CHANNELS == -1:
+            hid_channels = in_channels
+        else:
+            hid_channels = cfg.MODEL.PROPOSAL_GENERATOR.HIDDEN_CHANNELS
+        self.proposal_generator.rpn_head.conv = nn.Conv2d(
+            in_channels, hid_channels, kernel_size=3, stride=1, padding=1
+        )
+        shape = self.backbone.output_shape()
+        features = cfg.MODEL.RPN.IN_FEATURES
+        example_head = StandardRPNHead.from_config(cfg, [shape[f] for f in features])
+        num_cell_anchors = example_head["num_anchors"]
+        box_dim = example_head["box_dim"]
+        self.proposal_generator.rpn_head.objectness_logits = nn.Conv2d(
+            hid_channels, num_cell_anchors, kernel_size=1, stride=1
+        )
+        self.proposal_generator.rpn_head.anchor_deltas = nn.Conv2d(
+            hid_channels, num_cell_anchors * box_dim, kernel_size=1, stride=1
+        )
+
+    def _fix_res5_block(self, cfg):
+        res5_halve = cfg.MODEL.ROI_BOX_HEAD.RES5HALVE
+        if not res5_halve:
+            """
+            Modifications for VG in RoI heads:
+            1. Change the stride of conv1 and shortcut in Res5.Block1 from 2 to 1
+            2. Modifying all conv2 with (padding: 1 --> 2) and (dilation: 1 --> 2)
+            """
+            self.roi_heads.res5[0].conv1.stride = (1, 1)
+            self.roi_heads.res5[0].shortcut.stride = (1, 1)
+            for i in range(3):
+                self.roi_heads.res5[i].conv2.padding = (2, 2)
+                self.roi_heads.res5[i].conv2.dilation = (2, 2)
+
+    def forward_for_roi_head(self, features, proposal_boxes):
+        # assert not proposal_boxes[0].requires_grad
+        box_features = self.roi_heads._shared_roi_transform(features, proposal_boxes)
+        feature_pooled = box_features.mean(dim=[2, 3])  # pooled to 1x1
+        return feature_pooled
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
```

Comment:
This method is not required. Check how to load the checkpoint without using `from_pretrained`